### PR TITLE
Enhance Tracing with New Metrics, Renames, and Fixes

### PR DESCRIPTION
**Pull request by jutaro**
**Pull Request Description**

### Summary

This PR introduces several enhancements and bug fixes to the new tracing system, including two additional metrics, improved naming consistency, and bug fixes to align with the legacy tracing system.
These additions and fixes are introduced to:
- Enhance traceability by adding essential metrics and harmonizing names across legacy and new systems.
- Address outstanding bugs to ensure accuracy and reliability in the tracing data.
- Support workflows dependent on `served.block.latest` and other core metrics from the legacy system.

### New Metrics Added

1. **`node.start.time`**: Captures the node’s start time in POSIX seconds for machine use, providing a standardized timestamp for node initialization.

2. **`served.block.latest`**: A legacy metric previously missing from the new system, now incorporated to track the latest served block and ensure compatibility with established monitoring workflows.

### Changes and Fixes

- **Harmonized Metric Names**: Standardized metric naming conventions to align with legacy tracing, ensuring consistency across the system and simplifying transition efforts.
- **Bug Fixes**: Corrected several issues in the current metrics implementation:
    - Documentation for Churn Action duration metrics.
    - Fixes for metrics associated with forks, client operations, fetches, forging, and CDF counters.
    - Renamed P2P metrics for better alignment with naming conventions.
- **Client Tracer Rewrite**: Significant improvements in client tracer functionality to improve accuracy and consistency.

### CODE REVIEW

File: cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs [hunk 1]
```diff
@@ -540,23 +541,26 @@ instance ( LogFormatting (Header blk)
...
-    let ChainInformation { slots, blocks, density, epoch, slotInEpoch } =
-          chainInformation selChangedInfo newChain 0
+    let ChainInformation { .. } =
+          chainInformation selChangedInfo False newChain 0
     in  [ DoubleM "density" (fromRational density)
         , IntM    "slotNum" (fromIntegral slots)
         , IntM    "blockNum" (fromIntegral blocks)
         , IntM    "slotInEpoch" (fromIntegral slotInEpoch)
         , IntM    "epoch" (fromIntegral (unEpochNo epoch))
+        , CounterM "forks" (Just (if fork then 1 else 0))
```
[the remark this answers is not in the document]
Comment (jutaro) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs:563]: I like to keep it, is eaier for me to read this way. But I guess is a matter of taste.

File: cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs [hunk 2]
```diff
@@ -2096,21 +2100,24 @@ data ChainInformation = ChainInformation
...
 
 chainInformation
   :: forall blk. HasHeader (Header blk)
   => ChainDB.SelectionChangedInfo blk
+  -> Bool
   -> AF.AnchoredFragment (Header blk)
   -> Int64
   -> ChainInformation
-chainInformation selChangedInfo frag blocksUncoupledDelta = ChainInformation
+chainInformation selChangedInfo fork frag blocksUncoupledDelta = ChainInformation
```
[the remark this answers is not in the document]
Comment (jutaro) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs:2113]: fork is used for the "fork" counter metrics, so it is required as it is implemented now

File: cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs [hunk 3]
```diff
@@ -580,68 +583,81 @@ initialClientMetrics =
...
+                      , cmCdf3sVar  = snd3 updatedMetrics
+                      , cmCdf5sVar  = thd3 updatedMetrics
+                      , cmDelay     = realToFrac forgeDelay
+                      , cmBlockSize = getSizeInBytes blockSize
+                      , cmTraceVars = True
+                      , cmTraceIt   = True
+                      , cmSlotMap   = slotMap'' }
+
+    updateMetrics slotMap' _slotNo =
+      let updatedMetrics = updateCDFs 0 forgeDelay
```
[the remark this answers is not in the document]
Comment (jutaro) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs:628]: Done

File: cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs [hunk 4]
```diff
@@ -873,6 +890,48 @@ instance MetaTrace (TraceBlockFetchServerEvent blk) where
...
+    [IntM "served.block.latest" (fromIntegral servedBlocksLatest)]
+
+emptyServedBlocks :: ServedBlock
+emptyServedBlocks = ServedBlock 0 0 0
+
+servedBlockLatest ::
+     Maybe (Trace IO FormattedMessage)
+  -> IO (Trace IO  (TraceLabelPeer peer (TraceBlockFetchServerEvent blk)))
+servedBlockLatest mbTrEKG =
+      foldTraceM calculateServedBlockLatest emptyServedBlocks
```
[the remark this answers is not in the document]
Comment (jutaro) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs:917]: I could add a fold with a pure aggregation function, but I'm a bit undecided if it helps more or confuses more. So I would like to stay with the current lib for the moment

File: cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs [hunk 5]
```diff
@@ -873,6 +890,48 @@ instance MetaTrace (TraceBlockFetchServerEvent blk) where
...
+                  (metricsFormatter
+                    (mkMetricsTracer mbTrEKG))
+
+calculateServedBlockLatest :: ServedBlock
+                          -> LoggingContext
+                          -> TraceLabelPeer peer (TraceBlockFetchServerEvent blk)
+                          -> IO ServedBlock
```
[the remark this answers is not in the document]
Comment (jutaro) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs:924]: Done

File: cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs [hunk 6]
```diff
@@ -500,60 +501,62 @@ instance MetaTrace (TraceChainSyncServerEvent blk) where
...
+                          [ CounterM "blockfetchclient.lateblocks" Nothing
+                          | delay > 5
+                          ]
+        in msgs
+    else []
 
 instance MetaTrace ClientMetrics where
   namespaceFor _ = Namespace [] ["ClientMetrics"]
-  severityFor _ _ = Just Info
+  severityFor _ _ = Just Debug
```
[the remark this answers is not in the document]
Comment (mgmeier) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs:559]: Makes sense, thank you.

File: cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs (hunk 4 above)
[the remark this answers is not in the document]
Comment (mgmeier) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/Consensus.hs:914]: Yes. We can take another look at that - goes beyond the scope of this PR.

**Review by mgmeier — APPROVED** [review]
LGTM, thank you @jutaro

**Pull request merged**